### PR TITLE
Create person, age, and teacher class

### PR DESCRIPTION
**Pull request by Jolak5**
In this PR, I:

1. Created class Person
2. Create class Student
3. Create class Teacher

### CODE REVIEW

**Review by GedeonTS — COMMENTED** [review]
# INVALID REVIEW REQUEST

Hi @Jolak5 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has Linters failing. Kindly Fix that before a new submission

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Student < Person
+  attr_reader :classroom
+
+  def initialize(age, name = 'Unknown', classroom: '')
+    super(name, age)
```
Comment (ShadyShawkat) [inline student.rb]: - [ ] Please pass the `parent_permission` parameter to the `initialize` method which is then passed to the `super` method.

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Teacher < Person
+  attr_reader :specialization
+
+  def initialize(age, name = 'Unknown', specialization: '')
+    super(name, age)
+    @specialization = specialization
```
Comment (ShadyShawkat) [inline teacher.rb]: - [ ] Please pass the `parent_permission` parameter to the `initialize` method which is then passed to the `super` method.

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
```
Comment (ShadyShawkat) [inline person.rb]: - [ ] Kindly make this method private, as per [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=Private%20method%20of_age%3F) requirement.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @Jolak5, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 3 above)
Comment (codepantha) [inline person.rb]: - [ ] Just as suggested by the previous reviewer, kindly make this method private. It is a requirement as you can see [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=Private%20method%20of_age%3F%20that%20returns%20true%20if%20%40age%20is%20greater%20or%20equal%20to%2018%20and%20false%20otherwise.)

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Jolak5 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ Used GitFlow
✔️ No linter errors

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- NA
### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**